### PR TITLE
Restructured app

### PR DESCRIPTION
**Pull request by vorugantia**
Pull request contains: 
- Most recent dependencies in package.json
- Fully refactored and restructured app
- Just a (working) skeleton of the website - no Aladin map, etc.

@cdeil

### CODE REVIEW

**Comment by vorugantia** [comment]
I'll fix the travis-ci builds first...

**Comment by vorugantia** [comment]
Commented out spec.ts files for now, travis-ci builds passed.

File: .gitignore [hunk 1]
```diff
@@ -26,6 +30,4 @@ testem.log
 
 #System Files
 .DS_Store
-
-.vscode
-__pycache__
```
Comment (cdeil) [inline .gitignore:31]: Don't remove these two lines.

**Comment by cdeil** [comment]
Remove all `.pyc` files, they shouldn't be under version control.
E.g. `gammasky/__pycache__/__init__.cpython-35.pyc`

File: package.json [hunk 2]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "gamma-sky",
-  "version": "0.0.1",
+  "version": "0.0.0",
```
Comment (cdeil) [inline package.json]: Go to 0.0.2 here.

File: package.json [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
   "license": "MIT",
   "angular-cli": {},
   "scripts": {
-    "start": "ng server",
+    "start": "ng serve",
     "postinstall": "typings install",
     "lint": "tslint \"src/**/*.ts\"",
-    "format": "clang-format -i -style=file --glob=src/**/*.ts",
+    "test": "ng test",
```
Comment (cdeil) [inline package.json:10]: Maybe try running ng test also on travis-ci.
If it fails, OK to not run it this way.

File: e2e/app.e2e-spec.ts [hunk 4]
```diff
@@ -5,10 +5,10 @@ describe('gamma-sky App', function() {
 
   beforeEach(() => {
     page = new GammaSkyPage();
-  })
+  });
 
   it('should display message saying app works', () => {
     page.navigateTo();
-    expect(page.getParagraphText()).toEqual('gamma-sky works!');
+    expect(page.getParagraphText()).toEqual('app works!');
```
Comment (cdeil) [inline e2e/app.e2e-spec.ts]: Try running `ng e2e` locally. Does it pass? What are these? Should we run them on travis-ci?

File: src/app/views/cat-view/cat-view.component.html [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<p>
+  This is CatViewComponent
+</p>
+
+<switch-view [notCurrentView]="notCurrentView"></switch-view>
```
Comment (cdeil) [inline src/app/views/cat-view/cat-view.component.html]: Rename data member to "selectedView".

File: src/app/views/switch-view/switch-view.component.html [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<div>
+  <button id='switchViewButton' (click)='onClick($event)'>
+    <b>Go to {{notCurrentView}}</b>
```
Comment (cdeil) [inline src/app/views/switch-view/switch-view.component.html]: if selectedView == 'map':
    "go to cat"
...

File: src/app/app.routes.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import { provideRouter, RouterConfig } from '@angular/router';
+import { MapViewComponent } from './views/map-view/map-view.component';
+import { CatViewComponent } from './views/cat-view/cat-view.component';
+import { CatHelpComponent } from './widgets/cat-help/cat-help.component';
+import { CatSource2FHLComponent } from './widgets/cat-source/cat-source-2fhl/cat-source-2fhl.component';
+import { CatSource3FGLComponent } from './widgets/cat-source/cat-source-3fgl/cat-source-3fgl.component';
+import { CatSourceSNRcatComponent } from './widgets/cat-source/cat-source-snrcat/cat-source-snrcat.component';
```
Comment (cdeil) [inline src/app/app.routes.ts:7]: Could try this here in `src/app/app.routes.ts` (or from anywhere else):

```
import { CatSourceSNRcatComponent } from './widgets/cat-source';
```

And in `app/widgets/cat-source/index.ts` put this:

```
export { CatSourceSNRcatComponent } from './cat-source-snrcat/cat-source-snrcat.component'
```

**Comment by cdeil** [comment]
Add back data files.

Maybe: add back content of about component HTML and CSS

File: .gitignore (hunk 1 above)
Comment (vorugantia) [inline .gitignore]: done

Comment (vorugantia) [inline .gitignore:31]: done

File: package.json (hunk 2 above)
Comment (vorugantia) [inline package.json]: done

File: e2e/app.e2e-spec.ts (hunk 4 above)
Comment (vorugantia) [inline e2e/app.e2e-spec.ts]: Passes locally. I'll add all new travis-ci commands in one commit

File: src/app/views/cat-view/cat-view.component.html (hunk 5 above)
Comment (vorugantia) [inline src/app/views/cat-view/cat-view.component.html]: done

File: src/app/views/switch-view/switch-view.component.html (hunk 6 above)
Comment (vorugantia) [inline src/app/views/switch-view/switch-view.component.html]: done

File: src/app/app.routes.ts (hunk 7 above)
Comment (vorugantia) [inline src/app/app.routes.ts:7]: @cdeil - check out `system-config.ts`, this is where angular-cli generates all barrels. If we want to make our own barrels, I'll have to add them in myself in this file. However, there's a scary "Everything underneath this line is managed by the CLI" message so I'm not sure if I should manually configure barrels.

For this PR, I'm not going to do anything about the barrels and I'll just leave all of the generated `index.ts` files.

**Comment by vorugantia** [comment]
PR fixes:
- Added data
- Added `ng e2e` to travis-ci (`ng test` didn't work)
- Removed `.pyc` files, fixed .gitignore
- Added code for about button
- gamma-sky v0.0.2

Merging into master...

( @cdeil - see my note in `app.routes.ts` under yours, about barrels.)

**Comment by cdeil** [comment]
OK to not introduce other barrels for me. It's not really needed and if it's working now, maybe as you say best just keep as is.

Great progress today, good job!

See you tomorrow.

**Pull request merged**